src/cosmology/cosmology_functions.cpp

```diff
@@ -130,4 +174,64 @@ void Grid3D::Change_GAS_Frame_System(bool forward)
   }
 }
 
+/* local function that designates whether we are using a root-process. It gives
+ * gives a sensible result regardless of whether we are using MPI */
+static inline bool Is_Root_Proc()
+{
+  #ifdef MPI_CHOLLA
+  return procID == root;
+  #else
+  return true;
+  #endif
+}
+
+void Cosmology::Create_Expansion_History_File(struct Parameters *P)
+{
+  // writing the expansion history
+  if (not Is_Root_Proc()) {
+    return;
+  }
+
+  std::string file_name(EXPANSION_HISTORY_FILE_NAME);
+  chprintf("\nCreating Expansion History File: %s \n\n", file_name.c_str());
+
+  bool file_exists = false;
+  if (FILE *file = fopen(file_name.c_str(), "r")) {
+    file_exists = true;
+    chprintf("  File exists, appending values: %s \n\n", file_name.c_str());
+    fclose(file);
+  }
+
+  // current date/time based on current system
+  time_t now = time(0);
+  // convert now to string form
+  char *dt = ctime(&now);
+
+  std::string message = "# H0 OmegaM Omega_b OmegaL w0 wa Omega_R Omega_K\n";
+  message += "# " + std::to_string(H0 * 1e3) + " " + std::to_string(Omega_M);
+  message += " " + std::to_string(Omega_b);
+  message += " " + std::to_string(Omega_L) + " " + std::to_string(w0) + " " + std::to_string(wa);
+  message += " " + std::to_string(Omega_R) + " " + std::to_string(Omega_K);
+
+  std::ofstream out_file;
+  out_file.open(file_name.c_str(), std::ios::app);
+  out_file << "# Run date: " << dt;
+  out_file << message.c_str() << std::endl;
+  out_file.close();
+}
+
+void Cosmology::Write_Expansion_History_Entry(void)
```

Comment:
```suggestion
/* Write the current entry to the expansion history file */
void Cosmology::Write_Expansion_History_Entry(void)
```

Reply:
I'll default to this format in the future.